### DISCUSSION TITLE
Compile a charon plugin from outside of the strongswan directory

### DISCUSSION THREAD

**Discussion by MOHITCHITLANGE** [strongswan/strongswan#368] [Q&A]
I am writing an plugin and I want that plugin, but I want source code of that plugin to be picked up from the location mentioned. Is there a way to get this done ?

**Comment by tobiasbrunner** [comment]
You can build plugins out of the strongSwan source tree. See [this example plugin](https://github.com/tobiasbrunner/strongswan-redirect-plugin).
